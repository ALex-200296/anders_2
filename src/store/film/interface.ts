import { FilmTypes } from "./types";

export interface IFilmList {
  Poster: string;
  Title: string;
  Type?: string;
  Year: string;
  imdbID: string;
}

export interface IFilmItem {
  Actors?: string;
  Awards?: string;
  BoxOffice?: string;
  Country?: string;
  Director?: string;
  Language?: string;
  Plot?: string;
  Poster?: string;
  Production?: string;
  Released?: string;
  Title?: string;
  Writer?: string;
  Year?: string;
  imdbID?: string;
  imdbRating?: string;
  imdbVotes?: string;
}

export interface IFilmInitialState {
  search: string;
  data: IFilmList[];
  error: string;
  loading: boolean;
  item: IFilmItem;
}

export interface IFilmAction {
  type: FilmTypes.FETCH_FILMS;
}

export interface IFilmActionError {
  type: FilmTypes.FETCH_FILMS_ERROR;
  error: string;
}

export interface IFilmActionSuccess {
  type: FilmTypes.FETCH_FILMS_SUCCESS;
  data: [];
}

export interface IFilmActionID {
  type: FilmTypes.FETCH_FILM_ID;
  item: {};
}

export type FilmAction =
  | IFilmAction
  | IFilmActionError
  | IFilmActionSuccess
  | IFilmActionID
  | any;
