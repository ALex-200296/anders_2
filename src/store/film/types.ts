/* eslint-disable no-unused-vars */
export enum FilmTypes {
  FETCH_FILMS = "FETCH_FILMS",
  FETCH_FILMS_ERROR = "FETCH_FILMS_ERROR",
  FETCH_FILMS_SUCCESS = "FETCH_FILMS_SUCCESS",
}
