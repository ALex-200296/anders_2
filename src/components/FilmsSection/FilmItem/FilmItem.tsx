import React from "react";
import styles from "./filmItem.module.scss";

const FilmItem = () => {
  return <li className={styles.item}>hello </li>;
};

export default FilmItem;
