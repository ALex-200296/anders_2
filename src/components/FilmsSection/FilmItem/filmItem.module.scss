@import "../../../App.scss";

.item {
  position: relative;
  display: flex;
  flex-direction: column;

  border: 1px solid $grayE6;
  border-radius: 15px;
  -webkit-box-shadow: 0px 0px 28px 7px rgba(34, 60, 80, 0.2);
  -moz-box-shadow: 0px 0px 28px 7px rgba(34, 60, 80, 0.2);
  box-shadow: 0px 0px 28px 7px rgba(34, 60, 80, 0.2);
}

.item:hover,
.item:focus {
  background-color: $lightBlue;
}

.title {
  text-align: center;
}

.img {
  max-width: 100%;
  width: 100%;
  height: 400px;
  object-fit: contain;
}

.details {
  display: flex;
  justify-content: center;
}

.link {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}
