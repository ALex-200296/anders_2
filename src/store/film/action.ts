import { IFilmAction, IFilmActionError, IFilmActionSuccess } from "./interface";
import { FilmTypes } from "./types";

export const filmAction = (): IFilmAction => ({
  type: FilmTypes.FETCH_FILMS,
});

export const filmActionError = (error: string): IFilmActionError => ({
  type: FilmTypes.FETCH_FILMS_ERROR,
  error: error,
});

export const filmActionSuccess = (data: []): IFilmActionSuccess => ({
  type: FilmTypes.FETCH_FILMS_SUCCESS,
  data: data,
});
