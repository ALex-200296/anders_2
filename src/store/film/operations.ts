import axios from "axios";
import { Dispatch } from "react";
import { RouteComponentProps } from "react-router-dom";
import { RoutesFilm } from "../../pages/Home/routes/homeRoutesConfig";
import { RoutesHome } from "../../routes/routesConfig";
import {
  filmAction,
  filmActionError,
  filmActionID,
  filmActionSuccess,
} from "./action";
import { FilmAction } from "./interface";

interface IData {
  Error: string;
  Response: string;
  Search: [];
}

const { REACT_APP_FILMS, REACT_APP_API_KEY } = process.env;

export const fetchOperationFilms = (
  search: string = "",
  history?: RouteComponentProps["history"]
) => {
  return async (dispatch: Dispatch<FilmAction>) => {
    try {
      dispatch(filmAction());
      if (search) {
        const data: IData = await axios
          .get(`${REACT_APP_FILMS}`, {
            params: {
              apikey: REACT_APP_API_KEY,
              s: search,
            },
          })
          .then(({ data }) => data);
        if (data.Response === "True") {
          console.log(data.Search);
          dispatch(filmActionSuccess(data.Search));
          history?.push(RoutesFilm);
        }
        if (data.Response === "False") {
          dispatch(filmActionError(data.Error));
          history?.push(RoutesHome);
        }
      }
    } catch (e) {
      console.log(e);
    }
  };
};

export const fetchOperationFilmID = (id: string) => {
  return async (dispatch: Dispatch<FilmAction>) => {
    try {
      dispatch(filmAction());
      const data = await axios
        .get(`${REACT_APP_FILMS}`, {
          params: {
            apikey: REACT_APP_API_KEY,
            i: id,
          },
        })
        .then(({ data }) => data);

      dispatch(filmActionID(data));
    } catch (e) {
      console.log(e);
    }
  };
};
