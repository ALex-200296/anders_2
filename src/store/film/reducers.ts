import { FilmAction, IFilmInitialState } from "./interface";
import { FilmTypes } from "./types";

export const initialState: IFilmInitialState = {
  data: [],
  error: "",
  loading: false,
  search: "",
  item: {},
};

export const filmReducer = (state = initialState, action: FilmAction) => {
  switch (action.type) {
    case FilmTypes.FETCH_FILMS:
      return { ...state, loading: true };
    case FilmTypes.FETCH_FILMS_ERROR:
      return { ...state, loading: false, error: action.error };
    case FilmTypes.FETCH_FILMS_SUCCESS:
      return { ...state, loading: false, data: action.data };
    case FilmTypes.FETCH_FILM_ID:
      return { ...state, item: action.item, loading: false };
    default:
      return state;
  }
};
