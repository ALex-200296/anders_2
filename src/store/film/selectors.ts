export const getFilmDataSelector = (state: any) => state.film;
