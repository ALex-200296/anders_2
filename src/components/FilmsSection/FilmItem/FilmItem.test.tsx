import React from "react";
import { IFilmList } from "../../../store/film/interface";
import FilmItem from "./FilmItem";
import { render, shallow } from "enzyme";
import { MemoryRouter } from "react-router";

const data: IFilmList = {
  Poster: "this is images",
  Title: "Человек паук",
  Year: "15.06.2016",
  imdbID: "12345",
};

const setUp = (props?: IFilmList) =>
  render(
    <MemoryRouter>
      <FilmItem {...props} />
    </MemoryRouter>
  );

describe("FilmItem component", () => {
  it("should render FilmItem to props", () => {
    const component = setUp(data);
    expect(component).toMatchSnapshot();
  });
  it("should render FilmItem without props", () => {
    const component = setUp();
    expect(component).toMatchSnapshot();
  });
});

describe("FilmItem component", () => {
  it("FilmItem check for item li", () => {
    const component = shallow(<FilmItem />).find("li");
    expect(component.length).toBe(1);
  });
});
