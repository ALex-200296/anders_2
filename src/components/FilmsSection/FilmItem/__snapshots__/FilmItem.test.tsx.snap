// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmItem component should render FilmItem to props 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "item",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "title",
        },
        "children": Array [
          Node {
            "data": "Человек паук",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h3",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "alt": "this is images",
            "class": "img",
            "src": "this is images",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "class": "details",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "datetime": "15.06.2016",
                },
                "children": Array [
                  Node {
                    "data": "Год: 15.06.2016",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "time",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "datetime": undefined,
                },
                "x-attribsPrefix": Object {
                  "datetime": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "link",
                "href": "/film/12345",
              },
              "children": Array [],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "href": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "href": undefined,
              },
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "alt": undefined,
            "class": undefined,
            "src": undefined,
          },
          "x-attribsPrefix": Object {
            "alt": undefined,
            "class": undefined,
            "src": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "alt": "this is images",
          "class": "img",
          "src": "this is images",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "details",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "datetime": "15.06.2016",
              },
              "children": Array [
                Node {
                  "data": "Год: 15.06.2016",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "time",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "datetime": undefined,
              },
              "x-attribsPrefix": Object {
                "datetime": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "class": "link",
              "href": "/film/12345",
            },
            "children": Array [],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "href": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "href": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "title",
          },
          "children": Array [
            Node {
              "data": "Человек паук",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h3",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "alt": undefined,
          "class": undefined,
          "src": undefined,
        },
        "x-attribsPrefix": Object {
          "alt": undefined,
          "class": undefined,
          "src": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "details",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "datetime": "15.06.2016",
            },
            "children": Array [
              Node {
                "data": "Год: 15.06.2016",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "time",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "datetime": undefined,
            },
            "x-attribsPrefix": Object {
              "datetime": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "link",
            "href": "/film/12345",
          },
          "children": Array [],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "href": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "href": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "alt": "this is images",
            "class": "img",
            "src": "this is images",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "class": "title",
            },
            "children": Array [
              Node {
                "data": "Человек паук",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h3",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "alt": undefined,
            "class": undefined,
            "src": undefined,
          },
          "x-attribsPrefix": Object {
            "alt": undefined,
            "class": undefined,
            "src": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "link",
          "href": "/film/12345",
        },
        "children": Array [],
        "name": "a",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "details",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "datetime": "15.06.2016",
              },
              "children": Array [
                Node {
                  "data": "Год: 15.06.2016",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "time",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "datetime": undefined,
              },
              "x-attribsPrefix": Object {
                "datetime": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "alt": "this is images",
              "class": "img",
              "src": "this is images",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "title",
              },
              "children": Array [
                Node {
                  "data": "Человек паук",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h3",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "href": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "href": undefined,
        },
      },
    ],
    "name": "li",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`FilmItem component should render FilmItem without props 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "item",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "title",
        },
        "children": Array [
          Node {
            "data": "not found",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h3",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "alt": "not found",
            "class": "img",
            "src": "notimages.png",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "class": "details",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "datetime": "not found",
                },
                "children": Array [
                  Node {
                    "data": "Год: not found",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "time",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "datetime": undefined,
                },
                "x-attribsPrefix": Object {
                  "datetime": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "link",
                "href": "/film//",
              },
              "children": Array [],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "href": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "href": undefined,
              },
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "alt": undefined,
            "class": undefined,
            "src": undefined,
          },
          "x-attribsPrefix": Object {
            "alt": undefined,
            "class": undefined,
            "src": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "alt": "not found",
          "class": "img",
          "src": "notimages.png",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "details",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "datetime": "not found",
              },
              "children": Array [
                Node {
                  "data": "Год: not found",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "time",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "datetime": undefined,
              },
              "x-attribsPrefix": Object {
                "datetime": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "class": "link",
              "href": "/film//",
            },
            "children": Array [],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "href": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "href": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "title",
          },
          "children": Array [
            Node {
              "data": "not found",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h3",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "alt": undefined,
          "class": undefined,
          "src": undefined,
        },
        "x-attribsPrefix": Object {
          "alt": undefined,
          "class": undefined,
          "src": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "details",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "datetime": "not found",
            },
            "children": Array [
              Node {
                "data": "Год: not found",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "time",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "datetime": undefined,
            },
            "x-attribsPrefix": Object {
              "datetime": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "link",
            "href": "/film//",
          },
          "children": Array [],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "href": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "href": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "alt": "not found",
            "class": "img",
            "src": "notimages.png",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "class": "title",
            },
            "children": Array [
              Node {
                "data": "not found",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h3",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "alt": undefined,
            "class": undefined,
            "src": undefined,
          },
          "x-attribsPrefix": Object {
            "alt": undefined,
            "class": undefined,
            "src": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "link",
          "href": "/film//",
        },
        "children": Array [],
        "name": "a",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "details",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "datetime": "not found",
              },
              "children": Array [
                Node {
                  "data": "Год: not found",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "time",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "datetime": undefined,
              },
              "x-attribsPrefix": Object {
                "datetime": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "alt": "not found",
              "class": "img",
              "src": "notimages.png",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "title",
              },
              "children": Array [
                Node {
                  "data": "not found",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h3",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "href": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "href": undefined,
        },
      },
    ],
    "name": "li",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
